modules/SofaConstraint/src/SofaConstraint/GenericConstraintSolver.cpp

```diff
@@ -292,13 +296,48 @@ bool GenericConstraintSolver::buildSystem(const core::ConstraintParams *cParams,
         sofa::helper::AdvancedTimer::stepBegin("Get Compliance");
         msg_info() <<" computeCompliance in "  << constraintCorrections.size()<< " constraintCorrections" ;
 
-        for (unsigned int i=0; i<constraintCorrections.size(); i++)
-        {
-            core::behavior::BaseConstraintCorrection* cc = constraintCorrections[i];
-            if (!cc->isActive()) continue;
-            sofa::helper::AdvancedTimer::stepBegin("Object name: " + cc->getName());
-            cc->addComplianceInConstraintSpace(cParams, &current_cp->W);
-            sofa::helper::AdvancedTimer::stepEnd("Object name: " + cc->getName());
+        if(multithreading.getValue()){
+            helper::vector<std::thread> threads(constraintCorrections.size());
+            helper::vector<sofa::component::linearsolver::LPtrFullMatrix<double>> Ws;
+            Ws.resize(constraintCorrections.size());
+            helper::vector<core::ConstraintParams> Ps;
+            Ps.resize(constraintCorrections.size());
+
+            for (unsigned int i=0; i<constraintCorrections.size(); i++)
+            {
+                Ws[i].resize(current_cp->W.rowSize(),current_cp->W.colSize());
+                Ps[i] = *cParams;
+                auto Wi = &Ws[i];
+                auto Pi = Ps[i];
+                core::behavior::BaseConstraintCorrection* cc = constraintCorrections[i];
+                threads[i] = std::thread([Pi, Wi, cc] {
+                    if (cc->isActive()) {
```

Comment:
Maybe just get this condition out of the lambda function. It will avoid starting a thread for no reason when the cc isn't activated.